### ISSUE TITLE
Magento 2.4.4-p2 

### ISSUE THREAD

**Issue by kestraly** [cachewerk/magento-relay#2]
Relay has been installed on the server, after adding the plugin and configuring env.php, I ended up with the following error when running setup:upgrade.

Class 'Relay\Relay' not found#0 vendor/magento/zendframework1/library/Zend/Cache.php(153): CacheWerk\Relay\Cache\Backend\Relay->__construct()

**Comment by tillkruss** [comment]
Is Relay installed in your CLI environment?

You can test it by running `php --ri relay`

**Comment by kestraly** [comment]
Relay Support => enabled
Relay Version => 0.6.2
Available cache => 33500288
Available serializers => php, json, igbinary, msgpack
Available compression => lzf, zstd, lz4
Binary UUID => 2fddbdf1-831d-4290-aba6-07b4d1581d12
Git SHA => a8fb6c4e60ac5afdcd41310ff13b1e8baa3867f5
Allocator => relay
License state => unknown
License memory cap => 0
License request id =>
relay.enabled => true
relay.key =>
relay.maxmemory => 33554432
relay.maxmemory_pct => 75
relay.eviction_policy => noeviction
relay.eviction_sample_keys => 128
relay.initial_readers => 128
relay.invalidation_poll_usec => 5
relay.pconnect_default => 1
relay.max_endpoint_dbs => 32
relay.loglevel => none
relay.logfile =>

**Comment by tillkruss** [comment]
Odd, can you post the full error stack trace to see which line it is?

**Comment by michael-grunder** [comment]
Possibly a silly question, but this is the error I would expect if the extension wasn't loaded at all.

Does relay show as being loaded via a call to `phpinfo()` on the web server?

**Comment by kestraly** [comment]
Morning all,

@michael-grunder - Relay is being loaded phpinfo(). 

```
Relay Support | enabled
Relay Version | 0.6.2
Available cache | 31181008
Available serializers | php, json, igbinary, msgpack
```

Re ran setup, re ran compilation and then enabled in env.php

Then tried to run bin/magento s:s:d and received the following;

```
Class 'Relay\Relay' not found#0 vendor/magento/zendframework1/library/Zend/Cache.php(153): CacheWerk\Relay\Cache\Backend\Relay->__construct()
#1 vendor/magento/zendframework1/library/Zend/Cache.php(94): Zend_Cache::_makeBackend()
#2 vendor/magento/framework/App/Cache/Frontend/Factory.php(172): Zend_Cache::factory()
#3 vendor/magento/framework/Cache/Frontend/Adapter/Zend.php(38): Magento\Framework\App\Cache\Frontend\Factory->Magento\Framework\App\Cache\Frontend\{closure}()
#4 vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php(121): Magento\Framework\Cache\Frontend\Adapter\Zend->__construct()
#5 vendor/magento/framework/ObjectManager/Factory/Dynamic/Developer.php(66): Magento\Framework\ObjectManager\Factory\AbstractFactory->createObject()
#6 vendor/magento/framework/ObjectManager/ObjectManager.php(56): Magento\Framework\ObjectManager\Factory\Dynamic\Developer->create()
#7 vendor/magento/framework/App/Cache/Frontend/Factory.php(174): Magento\Framework\ObjectManager\ObjectManager->create()
#8 vendor/magento/framework/App/Cache/Frontend/Pool.php(69): Magento\Framework\App\Cache\Frontend\Factory->create()
#9 vendor/magento/framework/App/Cache/Frontend/Pool.php(156): Magento\Framework\App\Cache\Frontend\Pool->_initialize()
#10 vendor/magento/framework/App/Cache/Type/FrontendPool.php(87): Magento\Framework\App\Cache\Frontend\Pool->get()
#11 vendor/magento/framework/App/Cache/Type/Config.php(49): Magento\Framework\App\Cache\Type\FrontendPool->get()
#12 vendor/magento/framework/Cache/Frontend/Decorator/Bare.php(65): Magento\Framework\App\Cache\Type\Config->_getFrontend()
#13 vendor/magento/framework/App/ObjectManager/ConfigLoader.php(73): Magento\Framework\Cache\Frontend\Decorator\Bare->load()
#14 vendor/magento/framework/App/ObjectManager/Environment/Developer.php(79): Magento\Framework\App\ObjectManager\ConfigLoader->load()
#15 vendor/magento/framework/App/ObjectManagerFactory.php(191): Magento\Framework\App\ObjectManager\Environment\Developer->configureObjectManager()
#16 vendor/magento/framework/App/Bootstrap.php(212): Magento\Framework\App\ObjectManagerFactory->create()
#17 vendor/magento/framework/App/Bootstrap.php(127): Magento\Framework\App\Bootstrap->__construct()
#18 vendor/magento/framework/Console/Cli.php(184): Magento\Framework\App\Bootstrap::create()
#19 vendor/magento/framework/Console/Cli.php(84): Magento\Framework\Console\Cli->initObjectManager()
#20 bin/magento(22): Magento\Framework\Console\Cli->__construct()
```

Many thanks

**Comment by tillkruss** [comment]
@kestraly So there might be 1-2 pitfalls here, let me elaborate.

First the `relay` extension might need to be installed for both your PHP FPM and your CLI environment. On some system they have different configurations and extensions unfortunately. In CLI you can run `php --ri relay` to see if it's working, while with FPM you need to create a `info.php` file with `<?php phpinfo();` in it and request it view HTTP to see if Relay was loaded.

Second, make sure you restart PHP after making configuration changes, like adding `extensions = relay.so`. The process must stop and start for new extensions to be recognized. 

Lastly, make sure that that all your PHP versions, if you have 8.0 and 8.1 for example on your system, have `relay` installed.
